### PR TITLE
Added documentation for richInspectVariables request

### PR DESCRIPTION
**Pull request by JohanMabille**
This PR adds documentation for a new debug requets in the protocol.

### CODE REVIEW

File: docs/messaging.rst [hunk 1]
```diff
@@ -1190,6 +1190,33 @@ The `inspectVariables` is meant to retrieve the values of all the variables that
           }
       }
 
+The `richInspectVariables` request allows to get the rich representation of a variable that has been defined in the kernel.
```
Comment (Carreau) [inline docs/messaging.rst:1193]:
```suggestion
The ``richInspectVariables`` request allows to get the rich representation of a variable that has been defined in the kernel.
```

File: docs/messaging.rst [hunk 2]
```diff
@@ -1190,6 +1190,33 @@ The `inspectVariables` is meant to retrieve the values of all the variables that
...
+
+  Content of the `richInspectVariables` request::
```
Comment (Carreau) [inline docs/messaging.rst:1195]:
```suggestion
  Content of the ``richInspectVariables`` request::
```

File: docs/messaging.rst [hunk 3]
```diff
@@ -1190,6 +1190,33 @@ The `inspectVariables` is meant to retrieve the values of all the variables that
...
+
+      {
+          'type' : 'request',
+          'command' : 'richInspectVariables,
```
Comment (Carreau) [inline docs/messaging.rst:1199]:
```suggestion
          'command' : 'richInspectVariables',
```

File: docs/messaging.rst [hunk 4]
```diff
@@ -1190,6 +1190,33 @@ The `inspectVariables` is meant to retrieve the values of all the variables that
...
+          'arguments' : {
+              # The variable name is used when the debugger is not stopped on a breakpoint.
+              'variableName' : str,
+              # The variable reference is used when the debugger hit a breakpoint.
+              'variableReference' : int
+          }
+      }
+
+  Content of the `richInspectVariables` response::
```
Comment (Carreau) [inline docs/messaging.rst:1208]:
```suggestion
  Content of the ``richInspectVariables`` response::
```

**Review by Carreau — APPROVED** [review]
missing quote, and I believe double backticks, I'll commit directly the changes to your branch, let me know if I was wrong, +1 to merge once CI gree.

**Comment by JohanMabille** [comment]
Hold on, I have a slight change to push.

**Comment by davidbrochart** [comment]
Double backticks are still missing for `inspectVariables`, `debugInfo` and `dumpCell`, but that can be done in a separate PR.

**Pull request merged**